platform/measurement_utils.cpp

```diff
@@ -44,7 +44,9 @@ bool FormatDistanceImpl(double m, string & res,
     res = ToStringPrecision(v, v >= 10.0 ? 0 : 1) + high;
   }
   else
-    res = ToStringPrecision(lowV, 0) + low;
+    // To display unit number only if < 100
+    // The explicit cast allows to force rounding of "lowV" with division and multiplication successive by 10.
+    res = ToStringPrecision(lowV > 100.0 ? (((int)lowV + 5) / 10) * 10 : lowV, 0) + low;
```

Comment:
1. Add the braces around this line, please.
2. We do not use C-style casts. You need static_cast here. Also, ```ceil``` may simplify the code.
3. The comment is about ```< 100``` while the code checks ```> 100```. It is hard to read (and contradictory in the case ```lowV == 100```).